File: src/ProjectR/content/scripts/spells/ScytheOfTheDoomed.lua
function GetTargetType()
    return Single
end

function GetName()
  return "Scythe of the Doomed"
end

function GetDescription()
    return "Slash at the life source of your target.\nChance to instantly kill target."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .32
end

function GetDelay()
  return .45
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  local damage = 2.7225 * aAD - .825 * dDEF
  d:ApplyDebuff(DTH, 40)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/ScytheOfTheDoomed.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class ScytheOfTheDoomed : SpellScriptBase
    {
        public override string Name { get { return "Scythe of the Doomed"; } }
        public override string Description { get { return "Slash at the life source of your target.\nChance to instantly kill target."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 32; } }
        public override double Delay { get { return .45; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = 2.7225 * aAD - .825 * dDEF;
            TryToApplyDebuff(DebuffResistance.DTH, 40);
            DealDamage(damage);
        }
    }
}